src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -1065,325 +1066,26 @@ t8_forest_element_face_normal (t8_forest_t forest, t8_locidx_t ltreeid, const t8
-      if (!is_inside[ipoint]) {
-        /* point_inside is true if the point was inside the first or second triangle. Otherwise it is false. */
-        is_inside[ipoint] = t8_triangle_point_inside (p_1, v, w, &points[ipoint * 3], tolerance);
-      }
-    }
-    return;
-  }
-  case T8_ECLASS_TRIANGLE: {
-    double p_0[3], p_1[3], p_2[3];
-
-    /* Compute the vertex coordinates of the triangle */
-    t8_forest_element_coordinate (forest, ltreeid, element, 0, p_0);
-    t8_forest_element_coordinate (forest, ltreeid, element, 1, p_1);
-    t8_forest_element_coordinate (forest, ltreeid, element, 2, p_2);
-    double v[3];
-    double w[3];
-    /* v = v - p_0 = p_1 - p_0 */
-    t8_vec_axpyz (p_0, p_1, v, -1);
-    /* w = w - p_0 = p_2 - p_0 */
-    t8_vec_axpyz (p_0, p_2, w, -1);
-
-    for (int ipoint = 0; ipoint < num_points; ipoint++) {
-      is_inside[ipoint] = t8_triangle_point_inside (p_0, v, w, &points[ipoint * 3], tolerance);
-    }
-    return;
-  }
-  case T8_ECLASS_TET:
-  case T8_ECLASS_HEX:
-  case T8_ECLASS_PRISM:
-  case T8_ECLASS_PYRAMID: {
-    /* For bilinearly interpolated volume elements, a point is inside an element
-     * if and only if it lies on the inner side of each face.
-     * The inner side is defined as the side where the outside normal vector does not
-     * point to.
-     * The point is on this inner side if and only if the scalar product of
-     * a point on the plane minus the point with the outer normal of the face
-     * is >= 0.
-     *
-     * In other words, let p be the point to check, n the outer normal and x a point
-     * on the plane, then p is on the inner side if and only if
-     *  <x - p, n> >= 0
-     */
-
-    const int num_faces = ts->t8_element_num_faces (element);
-    /* Assume that every point is inside of the element */
-    for (int ipoint = 0; ipoint < num_points; ipoint++) {
-      is_inside[ipoint] = 1;
-    }
-    for (int iface = 0; iface < num_faces; ++iface) {
-      double face_normal[3];
-      /* Compute the outer normal n of the face */
-      t8_forest_element_face_normal (forest, ltreeid, element, iface, face_normal);
-      /* Compute a point x on the face */
-      const int afacecorner = ts->t8_element_get_face_corner (element, iface, 0);
-      double point_on_face[3];
-      t8_forest_element_coordinate (forest, ltreeid, element, afacecorner, point_on_face);
-      for (int ipoint = 0; ipoint < num_points; ipoint++) {
-        const int is_inside_iface = t8_plane_point_inside (point_on_face, face_normal, &points[ipoint * 3]);
-        if (is_inside_iface == 0) {
-          /* Point is on the outside of face iface. Update is_inside */
-          is_inside[ipoint] = 0;
-        }
-      }
-    }
-    return;
-  }
-  default:
-    SC_ABORT_NOT_REACHED ();
-  }
+  t8_geometry_c *geometry = t8_cmesh_get_tree_geometry (cmesh, cgtreeid);
+  geometry->t8_geom_point_batch_inside_element (forest, ltreeid, element, points, num_points, is_inside, tolerance);
```

Comment:
How to calling this function `t8_geom_points_inside_element` ?

Reply:
I usually don't like to use the plural form, because they can be misleading so easily. the difference between
```t8_geom_points_inside_element``` and 
```t8_geom_point_inside_element``` is not that much

Reply:
Applied your changes after reading your following comments